From: Matthew Wilcox <matthew.r.wilcox@intel.com>
Date: Fri, 9 May 2014 22:45:08 -0400
Subject: NVMe: Update data structures for NVMe 1.2
Git-commit: 23372af15e638bf3ce0764554db3b5e58bf7ceb8
Patch-Mainline: v3.16
References: bsc#943458

Include changes from the current set of ratified Technical Proposals
for NVMe 1.2.

Signed-off-by: Matthew Wilcox <matthew.r.wilcox@intel.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 include/uapi/linux/nvme.h | 19 +++++++++++++++----
 1 file changed, 15 insertions(+), 4 deletions(-)

diff --git a/include/uapi/linux/nvme.h b/include/uapi/linux/nvme.h
index 0127591..521ca5a 100644
--- a/include/uapi/linux/nvme.h
+++ b/include/uapi/linux/nvme.h
@@ -27,7 +27,12 @@ struct nvme_id_power_state {
 	__u8			read_lat;
 	__u8			write_tput;
 	__u8			write_lat;
-	__u8			rsvd16[16];
+	__le16			idle_power;
+	__u8			idle_scale;
+	__u8			rsvd19;
+	__le16			active_power;
+	__u8			active_work_scale;
+	__u8			rsvd23[9];
 };
 
 enum {
@@ -46,7 +51,8 @@ struct nvme_id_ctrl {
 	__u8			mic;
 	__u8			mdts;
 	__u16			cntlid;
-	__u8			rsvd80[176];
+	__u32			ver;
+	__u8			rsvd84[172];
 	__le16			oacs;
 	__u8			acl;
 	__u8			aerl;
@@ -56,7 +62,9 @@ struct nvme_id_ctrl {
 	__u8			npss;
 	__u8			avscc;
 	__u8			apsta;
-	__u8			rsvd266[246];
+	__le16			wctemp;
+	__le16			cctemp;
+	__u8			rsvd270[242];
 	__u8			sqes;
 	__u8			cqes;
 	__u8			rsvd514[2];
@@ -145,7 +153,10 @@ struct nvme_smart_log {
 	__u8			unsafe_shutdowns[16];
 	__u8			media_errors[16];
 	__u8			num_err_log_entries[16];
-	__u8			rsvd192[320];
+	__le32			warning_temp_time;
+	__le32			critical_comp_time;
+	__le16			temp_sensor[8];
+	__u8			rsvd216[296];
 };
 
 enum {
-- 
1.8.5.6

